src/analysis/ffi_type.rs

```diff
@@ -74,7 +74,7 @@ pub fn ffi_type(env: &Env, tid: TypeId, c_type: &str) -> Result {
 
 fn ffi_inner(env: &Env, tid: TypeId, inner: &str) -> Result {
     let typ = env.library.type_(tid);
-    match *typ {
+    match typ {
```

Comment:
Why this change? Seems unrelated. Also below :)

Reply:
Just adds noise and should really be in a separate commit at least.

Reply:
As mentioned in the PR description:

> * Treat strings in `Fundamental::Vulkan` as `&'static str` so that all this borrowing churn isn't necessary (not possible if we parse types from G-IR);
>   OTOH some of the `*` and `ref` removal is actually quite a nice simplification!

If we end up going ahead with this change I'll do it in a separate commit :). This was a rushed initial implementation to get the `Things to discuss:` list up somewhere with related diff context, should have mentioned that more clearly.

Reply:
> Why this change?

AS for the _why_: All these `*typ` move by value, and require `typ` to be `Copy`, which is not the case anymore thanks to a new variant `Vulkan(String)`.